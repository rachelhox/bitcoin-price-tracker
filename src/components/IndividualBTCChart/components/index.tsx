import CandleStickChart from './CandleStickChart';

export { CandleStickChart };
