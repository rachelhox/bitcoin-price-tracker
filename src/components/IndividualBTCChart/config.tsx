export const currencyKeys = ['usd'].sort();

//   export const getCurrencyParams = (key: string) => {
//     return networkParams[key] ?? {};
//   };
