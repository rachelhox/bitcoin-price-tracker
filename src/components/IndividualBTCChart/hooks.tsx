/* eslint-disable no-undef */
/* eslint-disable array-callback-return */
/* eslint-disable no-console */
import { useState, useEffect } from 'react';
import axios from 'axios';
import useSWR from 'swr';
import { useGeneralContext } from '@contexts';

export const useIndividualChartkHook = () => {
  const { currency } = useGeneralContext();
  const [candleChartData, setCandleChartData] = useState<Array<OHLCProps>>([]);
  const [period, setPeriod] = useState(90);
  const fetcher = (url: string) => axios.get(url).then((res) => res.data);
  // eslint-disable-next-line no-unused-vars
  const { data: chartData, isValidating } = useSWR(
    `${process.env.NEXT_PUBLIC_COINGECKO_API_URL}/coins/bitcoin/ohlc?vs_currency=${currency.key}&days=${period}`,
    fetcher,
    { refreshInterval: 60000 }
  );

  useEffect(() => {
    setCandleChartData([]);
    if (chartData && period === 90) {
      chartData.map((candle: any) => {
        const dateObj = new Date(candle[0]);
        const day = dateObj.getDate();
        const month = dateObj.getMonth() + 1;
        const year = dateObj.getFullYear();
        setCandleChartData((prev) => [
          ...prev,
          {
            x: new Date(year, month, day),
            open: candle[1],
            high: candle[2],
            low: candle[3],
            close: candle[4],
          },
        ]);
      });
    } else if (chartData) {
      chartData.map((candle: any) => {
        const dateObj = new Date(candle[0]);
        const day = dateObj.getDate();
        const month = dateObj.getMonth() + 1;
        const year = dateObj.getFullYear();
        const hour = dateObj.getHours();
        const minute = dateObj.getMinutes();
        setCandleChartData((prev) => [
          ...prev,
          {
            x: new Date(year, month, day, hour, minute),
            open: candle[1],
            high: candle[2],
            low: candle[3],
            close: candle[4],
          },
        ]);
      });
    }
  }, [chartData, isValidating, currency, period]);

  return { currency, period, setPeriod, candleChartData };
};
