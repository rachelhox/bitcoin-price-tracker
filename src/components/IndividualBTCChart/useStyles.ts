import { css } from '@mui/material';

const useStyles = () => {
  return {
    root: css({}),
  };
};

export default useStyles;
