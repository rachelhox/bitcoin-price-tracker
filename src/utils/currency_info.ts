interface IStringIndex extends Record<string, any> {}

/**
 * Takes the key name and returns the location. Will return null if undefined
 */

const currencies: IStringIndex = {
  usd: {
    name: 'US Dollar',
    key: 'usd',
    unit: '$',
    type: 'fiat',
  },
  aed: {
    name: 'United Arab Emirates Dirham',
    key: 'aed',
    unit: 'DH',
    type: 'fiat',
  },
  eur: {
    name: 'Euro',
    key: 'eur',
    unit: '€',
    type: 'fiat',
  },
  aud: {
    name: 'Australian Dollar',
    key: 'aud',
    unit: 'A$',
    type: 'fiat',
  },
  cad: {
    name: 'Canadian Dollar',
    key: 'cad',
    unit: 'CA$',
    type: 'fiat',
  },
  chf: {
    name: 'Swiss Franc',
    key: 'chf',
    unit: 'Fr.',
    type: 'fiat',
  },
  gbp: {
    name: 'British Pound Sterling',
    key: 'gbp',
    unit: '£',
    type: 'fiat',
  },
  hkd: {
    name: 'Hong Kong Dollar',
    key: 'hkd',
    unit: 'HK$',
    type: 'fiat',
  },
  sgd: {
    name: 'Singapore Dollar',
    key: 'sgd',
    unit: 'S$',
    type: 'fiat',
  },
  cny: {
    name: 'Chinese Yuan',
    key: 'cny',
    unit: '¥',
    type: 'fiat',
  },
  jpy: {
    name: 'Japanese Yen',
    key: 'jpy',
    unit: '¥',
    type: 'fiat',
  },
  inr: {
    name: 'Indian Rupee',
    key: 'inr',
    unit: '₹',
    type: 'fiat',
  },
  krw: {
    name: 'South Korean Won',
    key: 'krw',
    unit: '₩',
    type: 'fiat',
  },
};

export const getCurrencyInfo = (key: string | number) => {
  return currencies[key] ?? {};
};
