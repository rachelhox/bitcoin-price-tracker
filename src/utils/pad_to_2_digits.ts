/* eslint-disable */

export const padTo2Digits = (num: any) => {
  return String(num).padStart(2, '0');
};
