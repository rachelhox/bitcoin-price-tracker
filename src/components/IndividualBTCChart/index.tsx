/* eslint-disable no-unused-vars */
import { Box, Typography } from '@mui/material';
import useStyles from './useStyles';
import { CandleStickChart } from './components';
import { useIndividualChartkHook } from './hooks';

const IndividualBTCChart = () => {
  const styles = useStyles();
  const { candleChartData } = useIndividualChartkHook();

  return (
    <Box css={styles.root}>
      <Box css={styles.maxWidth}>
        <Typography css={styles.h6} variant="h6">
          Please select currency
        </Typography>
        <CandleStickChart candleChartData={candleChartData} />
      </Box>
    </Box>
  );
};

export default IndividualBTCChart;
