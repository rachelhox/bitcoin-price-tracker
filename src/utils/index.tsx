import { getCurrencyInfo } from './currency_info';
import { convertToMoney } from './convert_to_money';
import { padTo2Digits } from './pad_to_2_digits';

export { getCurrencyInfo, convertToMoney, padTo2Digits };
