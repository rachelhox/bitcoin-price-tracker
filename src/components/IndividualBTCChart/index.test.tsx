import React from 'react';
import renderer from 'react-test-renderer';
import { MockTheme } from '@tests/utils';
import IndividualBTCChart from '.';

// ==================================
// mocks
// ==================================

jest.mock('@components', () => ({
  TimePeriodButton: (props: any) => <div id="TimePeriodButton" {...props} />,
}));

jest.mock('./components', () => ({
  CandleStickChart: (props: any) => <div id="CandleStickChart" {...props} />,
}));

// ==================================
// unit tests
// ==================================
describe('component: IndividualBTCChart', () => {
  it('matches snapshot', () => {
    const component = renderer.create(
      <MockTheme>
        <IndividualBTCChart />
      </MockTheme>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  afterEach(() => {
    jest.clearAllMocks();
  });
});
