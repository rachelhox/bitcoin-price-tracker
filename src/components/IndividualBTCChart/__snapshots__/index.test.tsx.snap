// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component: IndividualBTCChart matches snapshot 1`] = `
<div
  className="MuiBox-root css-g1za54-IndividualBTCChart"
>
  <div
    className="MuiBox-root css-1bxp50g-IndividualBTCChart"
  >
    <div
      className="MuiBox-root css-1cj9h77-IndividualBTCChart"
    >
      <h1
        className="MuiTypography-root MuiTypography-h1 css-1yswwri-MuiTypography-root-IndividualBTCChart"
      >
        BTC/USD
      </h1>
      <div
        id="TimePeriodButton"
        period={90}
        setPeriod={[Function]}
      />
    </div>
    <div
      candleChartData={[]}
      id="CandleStickChart"
      period={90}
    />
  </div>
</div>
`;
